gprofiler/metadata/application_identifiers.py

```diff
@@ -0,0 +1,256 @@
+        )
+        if wsgi_arg is not _NON_AVAILABLE_ARG:
+            return f"uwsgi: {_append_python_module_to_proc_wd(process, wsgi_arg)}"
+
+        wsgi_config = self._find_wsgi_from_config_file(process)
+        if wsgi_config is not None:
+            return f"uwsgi: {_append_python_module_to_proc_wd(process, wsgi_config)}"
+
+        _logger.warning(
+            f"{self.__class__.__name__} Couldn't find uwsgi wsgi module, both from cmdline and from config file",
+            cmdline=process.cmdline(),
+            no_extra_to_server=True,
+        )
+        return None
+
+
+class _CeleryApplicationIdentifier(_ApplicationIdentifier):
+    @staticmethod
+    def is_celery_process(process: Process) -> bool:
+        if "celery" == os.path.basename(_get_cli_arg_by_index(process.cmdline(), 0)) or "celery" == os.path.basename(
+            _get_cli_arg_by_index(process.cmdline(), 1)
+        ):
+            return True
+
+        return _is_python_m_proc(process) and process.cmdline()[2] == "celery"
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        if not self.is_celery_process(process):
+            return None
+
+        app_name = _get_cli_arg_by_name(process.cmdline(), "-A") or _get_cli_arg_by_name(
+            process.cmdline(), "--app", check_for_equals_arg=True
+        )
+        if app_name is None:
+            _logger.warning(
+                f"{self.__class__.__name__}: Couldn't find positional argument -A or --app for application indication",
+                cmdline=process.cmdline(),
+                no_extra_to_server=True,
+            )
+            return None
+
+        return f"celery: {_append_python_module_to_proc_wd(process, app_name)}"
+
+
+class _PySparkApplicationIdentifier(_ApplicationIdentifier):
+    @staticmethod
+    def _is_pyspark_process(process: Process) -> bool:
+        # We're looking for pythonXX -m pyspark.daemon
+        return _is_python_m_proc(process) and process.cmdline()[2] == "pyspark.daemon"
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        # TODO: detect application name from parent java native spark process.
+        return "pyspark" if self._is_pyspark_process(process) else None
+
+
+class _PythonModuleApplicationIdentifier(_ApplicationIdentifier):
+    @staticmethod
+    def is_python_app(process: Process) -> bool:
+        if not _is_python_bin(_get_cli_arg_by_index(process.cmdline(), 0)):
+            return False
+
+        return _is_python_m_proc(process) or (len(process.cmdline()) >= 2 and process.cmdline()[1].endswith(".py"))
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        if not _is_python_bin(_get_cli_arg_by_index(process.cmdline(), 0)):
+            return None
+
+        module_arg = _get_cli_arg_by_name(process.cmdline(), "-m")
+        if module_arg is not _NON_AVAILABLE_ARG:
+            return f"python: {_append_python_module_to_proc_wd(process, module_arg)}"
```

Comment:
This doesn't make much sense. If I have, for example, `gprofiler` installed and I run `python3 -m gprofiler`, I'm going to get it as if `gprofiler.py` is in my cwd.

uwsgi/gunicorn et al, do not take it from the Python path. `python` does. So here we want to skip translation I suppose.

Reply:
kk I don't really care I think it's almost the same but agree

Reply:
It's not the same.... it will be shown very badly.